app/comparator.py

```
@@ -0,0 +1,134 @@
+            ComparisonResult containing diff content
+        """
+        try:
+            # Run the comparison script
+            result = subprocess.run(
+                [self.script_path, repo_path, old_commit, new_commit, repo_name],
+                capture_output=True,
+                text=True
+            )
+            
+            if result.returncode != 0:
+                raise Exception(f"Script failed: {result.stderr}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)